src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -849,7 +990,7 @@ public bool TryReplaceIndexes(string oldIndexName, string newIndexName, bool imm
             bool lockTaken = false;
             try
             {
-                Monitor.TryEnter(_indexAndTransformerLocker, 16, ref lockTaken);
+                Monitor.TryEnter(_locker, 16, ref lockTaken);
```

Comment:
Why do we try to protect database record in raft? It is already not safe to do so, since the changes can come from multiple nodes.